### PR TITLE
This month in mathlib --- Dec 2021

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: posts/month-in-mathlib-dec-2021.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* [PR #10808](https://github.com/leanprover-community/mathlib/pull/10808) :: feat(topology/metric/hausdorff_distance): more properties of cthickening
+* [PR #10322](https://github.com/leanprover-community/mathlib/pull/10322) :: feat(topology/continuous_function/bounded): continuous bounded real-valued functions form a normed vector lattice
+* [PR #10477](https://github.com/leanprover-community/mathlib/pull/10477) :: feat(measure_theory/integral): `∫ x in b..b+a, f x = ∫ x in c..c + a, f x` for a periodic `f`
+* [PR #9836](https://github.com/leanprover-community/mathlib/pull/9836) :: feat(analysis/normed_space/weak_dual): add polar sets in preparation for Banach-Alaoglu theorem
+* [PR #10625](https://github.com/leanprover-community/mathlib/pull/10625) :: feat(probability_theory/martingale): define martingales
+* [PR #10710](https://github.com/leanprover-community/mathlib/pull/10710) :: feat(probability_theory/martingale): add super/sub-martingales
+* [PR #10509](https://github.com/leanprover-community/mathlib/pull/10509) :: feat(combinatorics/additive/salem_spencer): Salem-Spencer sets and Roth numbers
```
Comment (YaelDillies) [inline posts/month-in-mathlib-dec-2021.md:65]: This is combinatorics.

**Comment by arthurpaulino** [comment]
https://github.com/leanprover-community/mathlib/pull/5698 was merged in December of January, depending on your timezone, but either is fine :D

**Comment by jcommelin** [comment]
> [leanprover-community/mathlib#5698](https://github.com/leanprover-community/mathlib/pull/5698) was merged in December of January, depending on your timezone, but either is fine :D

Thanks, I've added it to the list.

**Comment by arthurpaulino** [comment]
> Thanks, I've added it to the list.

Oh, since you actually added it, I think this description suits better: "Proves that the following graphs are strongly regular: the empty graph, the complete graph and the complement of a strongly regular graph"

File: posts/month-in-mathlib-dec-2021.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+type: text
+---
+
+December 2021 saw 565 commits to mathlib, which sets a new record.
+In this post we highlight some of these contributions.
+
+* Combinatorics.
+    - [PR #10287](https://github.com/leanprover-community/mathlib/pull/10287) defines graph colorings and partitions.
+    - [PR #10981](https://github.com/leanprover-community/mathlib/pull/10981) defines the concepts of walks, paths, cycles in graph theory.
+    - [PR #5698](https://github.com/leanprover-community/mathlib/pull/5698) introduces strong regularity conditions for graphs.
```
Comment (jcommelin) [inline posts/month-in-mathlib-dec-2021.md:20]:
```suggestion
    - [PR #5698](https://github.com/leanprover-community/mathlib/pull/5698) introduces strong regularity conditions for graphs and shows that complete graphs and complements of strongly regular graphs are strongly regular.
```

File: posts/month-in-mathlib-dec-2021.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    - [PR #9762](https://github.com/leanprover-community/mathlib/pull/9762)
+      defines simplicial complexes embedded in an ambient vector space or more generally in a module over an ordered ring.
+    - [PR #10927](https://github.com/leanprover-community/mathlib/pull/10927)
+      defines the alternating face map complex of a simplicial object.
+* Geometry
+    - [PR #10733](https://github.com/leanprover-community/mathlib/pull/10733)
+      shows that an integral scheme is reduced and irreducible.
+      That a scheme is reduced iff its stalks are reduced is shown in
+      [PR #10879](https://github.com/leanprover-community/mathlib/pull/10879).
+    - [PR #8611](https://github.com/leanprover-community/mathlib/pull/8611) defines the action of `SL(2, ℤ)` on the upper half plane, in preparation for modular forms.
```
Comment (hrmacbeth) [inline posts/month-in-mathlib-dec-2021.md:43]:
```suggestion
    - [PR #8611](https://github.com/leanprover-community/mathlib/pull/8611) defines the action of `SL(2, ℤ)` on the upper half plane and partly classifies its orbits.
```

File: posts/month-in-mathlib-dec-2021.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+type: text
+---
+
+December 2021 saw 565 commits to mathlib, which sets a new record.
+In this post we highlight some of these contributions.
+
+* Combinatorics.
+    - [PR #10287](https://github.com/leanprover-community/mathlib/pull/10287) defines graph colorings and partitions.
+    - [PR #10981](https://github.com/leanprover-community/mathlib/pull/10981) defines the concepts of walks, paths, cycles in graph theory.
+    - [PR #5698](https://github.com/leanprover-community/mathlib/pull/5698) introduces strong regularity conditions for graphs and shows that complete graphs and complements of strongly regular graphs are strongly regular.
```
Comment (arthurpaulino) [inline posts/month-in-mathlib-dec-2021.md:20]:
```suggestion
    - [PR #5698](https://github.com/leanprover-community/mathlib/pull/5698) proves that complete graphs, empty graphs and complements of strongly regular graphs are strongly regular.
```
We already had conditions for strongly regular graphs on `master` 👍🏼

File: posts/month-in-mathlib-dec-2021.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      [PR #10734](https://github.com/leanprover-community/mathlib/pull/10734) and
+      [PR #10775](https://github.com/leanprover-community/mathlib/pull/10775)
+      prove
+      that being reduced, finite or of finite type are local properties.
+    - [PR #9888](https://github.com/leanprover-community/mathlib/pull/9888)
+      does not introduce new mathematics, but introduces a new way that makes
+      it exceedingly easy to apply lemmas about group homomorphisms to a ring
+      homomorphism (or a linear map, or an algebra homomorphism, etc).
+* Algebraic topology.
+    - [PR #9762](https://github.com/leanprover-community/mathlib/pull/9762)
```
Comment (YaelDillies) [inline posts/month-in-mathlib-dec-2021.md:34]: I don't know where to put those and whether they are worth mentioning.
```
- [PR #10524] and [PR #10648] define two new kinds of convexity: star convexity and strict convexity. The latter should soon find use in analysis.
```

**Comment by kmill** [comment]
Maybe it's interesting how there's a topological counterexample in mathlib:
```
- [PR #11072](https://github.com/leanprover-community/mathlib/pull/11072) continuous bijections from a compact
  space to a T2 space are homeomorphisms, and gives a counterexample for when T2 is weakened to T1.
```

File: posts/month-in-mathlib-dec-2021.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      defines simplicial complexes embedded in an ambient vector space or more generally in a module over an ordered ring.
+    - [PR #10927](https://github.com/leanprover-community/mathlib/pull/10927)
+      defines the alternating face map complex of a simplicial object.
+* Geometry
+    - [PR #10733](https://github.com/leanprover-community/mathlib/pull/10733)
+      shows that an integral scheme is reduced and irreducible.
+      That a scheme is reduced iff its stalks are reduced is shown in
+      [PR #10879](https://github.com/leanprover-community/mathlib/pull/10879).
+    - [PR #8611](https://github.com/leanprover-community/mathlib/pull/8611) defines the action of `SL(2, ℤ)` on the upper half plane and partly classifies its orbits.
```
Comment (rwbarton) [inline posts/month-in-mathlib-dec-2021.md:43]:
```suggestion
    - [PR #8611](https://github.com/leanprover-community/mathlib/pull/8611) defines the action of `SL(2, ℤ)` on the upper half plane and partially classifies its orbits.
```

File: posts/month-in-mathlib-dec-2021.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+type: text
+---
+
+December 2021 saw 565 commits to mathlib, which sets a new record.
+In this post we highlight some of these contributions.
+
+* Combinatorics.
+    - [PR #10287](https://github.com/leanprover-community/mathlib/pull/10287) defines graph colorings and partitions.
+    - [PR #10981](https://github.com/leanprover-community/mathlib/pull/10981) defines the concepts of walks, paths, cycles in graph theory.
+    - [PR #10509](https://github.com/leanprover-community/mathlib/pull/10509) defines [Salem-Spencer sets](https://en.wikipedia.org/wiki/Salem%E2%80%93Spencer_set) and Roth numbers
```
Comment (robertylewis) [inline posts/month-in-mathlib-dec-2021.md:20]:
```suggestion
    - [PR #10509](https://github.com/leanprover-community/mathlib/pull/10509) defines [Salem-Spencer sets](https://en.wikipedia.org/wiki/Salem%E2%80%93Spencer_set) and Roth numbers.
```

File: posts/month-in-mathlib-dec-2021.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      as well as the notion of generic points and sober spaces. Then 
+      [PR #10989](https://github.com/leanprover-community/mathlib/pull/10989) and
+      [PR #11040](https://github.com/leanprover-community/mathlib/pull/11040)
+      show that schemes are sober.
+    - [PR #10701](https://github.com/leanprover-community/mathlib/pull/10701)
+      proves the [Tietze extension theorem](https://en.wikipedia.org/wiki/Tietze_extension_theorem).
+* Functional analysis
+    - [PR #10825](https://github.com/leanprover-community/mathlib/pull/10825) defines the adjoint of a continuous linear map between Hilbert spaces,
+      made possible by the [semilinear map refactor](https://leanprover-community.github.io/blog/posts/semilinear-maps).
+    - [PR #10837](https://github.com/leanprover-community/mathlib/pull/10837) shows that continuous linear maps on a Hilbert space form a $C^\ast$-algebra.
```
Comment (hrmacbeth) [inline posts/month-in-mathlib-dec-2021.md:60]: Can merge these, for efficiency:
```suggestion
    - [PR #10825](https://github.com/leanprover-community/mathlib/pull/10825) defines the adjoint of a continuous linear map between Hilbert spaces, made possible by the [semilinear map refactor](https://leanprover-community.github.io/blog/posts/semilinear-maps), and [PR #10837](https://github.com/leanprover-community/mathlib/pull/10837) uses this to put a $C^\ast$-algebra structure on the continuous linear maps on a Hilbert space.
```

**Pull request merged**